db/db_impl/db_impl.cc

```diff
@@ -627,12 +627,10 @@ Status DBImpl::CloseHelper() {
     mutex_.Lock();
   }
   if (!mutable_db_options_.avoid_sync_during_shutdown && !logs_.empty()) {
-    mutex_.Unlock();
-    Status s = SyncWAL();
-    mutex_.Lock();
+    Status s = FlushWAL(true /* sync */);
```

Comment:
Why is the mutex unlock/lock removed? I see that SyncWAL() might try to acquire mutex_ here: https://github.com/facebook/rocksdb/blob/62600cb2d4bf6392d871103cad5a31102c669bea/db/db_impl/db_impl.cc#L1650.

Reply:
A mistake of mine and didn't get caught in the UT without triggering that path

Reply:
Fixed